src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
As an above - having one `filter' call is a cleaner solution

Reply:
Please note - in this very case `some` should be preferred over `find` - you don't need to get the children of the person - you only check if the children are present. `some` gives better performance here.